test/game/ScoreLogic.spec.js

```diff
@@ -0,0 +1,59 @@
+import ava from 'ava';
+import sinon from 'sinon';
+import AircraftModel from '../../src/assets/scripts/client/aircraft/AircraftModel';
+import UiController from '../../src/assets/scripts/client/ui/UiController';
+import GameController, { GAME_EVENTS } from '../../src/assets/scripts/client/game/GameController';
+import ScoreLogic from '../../src/assets/scripts/client/game/ScoreLogic';
+import { ARRIVAL_AIRCRAFT_INIT_PROPS_MOCK } from '../aircraft/_mocks/aircraftMocks';
+
+let sandbox; // using the sinon sandbox ensures stubs are restored after each test
+
+/* eslint-disable no-unused-vars, no-undef */
+ava.beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+});
+
+ava.afterEach(() => {
+    sandbox.restore();
+});
+/* eslint-enable no-unused-vars, no-undef */
+
+ava('._penalizeLocalizerInterceptAltitude() records an event and notifies the user of their error when above the glideslope', (t) => {
+    const scoreLogic = new ScoreLogic();
+    const aircraftModel = new AircraftModel(ARRIVAL_AIRCRAFT_INIT_PROPS_MOCK);
+    const uiControllerUiLogStub = sandbox.stub(UiController, 'ui_log');
+    const gameControllerRecordEventStub = sandbox.stub(GameController, 'events_recordNew');
+    const expectedLogMessage = `${aircraftModel.getCallsign()} intercepted localizer above glideslope`;
+
+    sandbox.stub(aircraftModel, 'isAboveGlidepath', () => true);
+    scoreLogic._penalizeLocalizerInterceptAltitude(aircraftModel);
+
+    t.true(uiControllerUiLogStub.calledWithExactly(expectedLogMessage, true));
```

Comment:
if we aren't controlling the returns here, and are simply verifying call params, we should use `spies` instead of `stubs`.

Reply:
@n8rzz I tried to replace the stubs with spies but I get `Attempted to wrap ui_log which is already wrapped`. Maybe you can get it to work.

Reply:
you'll usually see a warning like that when some other test has stubbed a method.  the way to resolve that is to make sure that in any test where you _are_ using stubs, you also call `someMethodStub.restore()` after your assertions have been called.